GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs

```diff
@@ -115,7 +115,18 @@ public static string GetUniqueEnlistmentRootWithSpaces()
         public string GetObjectRoot(FileSystemRunner fileSystem)
         {
             IEnumerable<FileSystemInfo> localCacheRootItems = this.LocalCacheRoot.ShouldBeADirectory(fileSystem).WithItems();
-            localCacheRootItems.Count().ShouldEqual(2, "Expected local cache root to contain 2 items. Actual items: " + string.Join(",", localCacheRootItems));
+
+            FileInfo[] files = localCacheRootItems.Where(info => info is FileInfo).Cast<FileInfo>().ToArray();
+            files.Where(f => string.Equals(f.Name, "mapping.dat", StringComparison.OrdinalIgnoreCase))
+                 .Count()
+                 .ShouldEqual(1, "Local cache root should contain a single 'mapping.dat' file, actual files: " + string.Join<FileInfo>(",", files));
+            
+            IEnumerable<FileInfo> unexpectedFiles = files.Where(

```

Comment:
This looks fine to me, but you might consider creating an array (or maybe `HashSet<string>`, although it doesn't make a practical difference) of the list of expected file names up front, then you could simplify the LINQ statement to something like:

```
HashSet<string> expectedFiles = new HashSet<string>(StringComparer.OrdinalIgnoreCase) {"expected_file1", "expectedFile2"};
unexpectedFiles.Where(f => expectedFile.Contains(f.Name));
```